src/base/utils/gzip.cpp

```diff
@@ -27,116 +27,121 @@
  * exception statement from your version.
  */
 
+#include "gzip.h"
+
 #include <QByteArray>
-#include <zlib.h>
 
-#include "gzip.h"
+#define ZLIB_CONST  // make z_stream.next_in const
+#include <zlib.h>
 
-bool Utils::Gzip::compress(QByteArray src, QByteArray &dest)
+QByteArray Utils::Gzip::compress(const QByteArray &data, bool *ok, const int level)
 {
-    static const int BUFSIZE = 128 * 1024;
-    char tmpBuf[BUFSIZE];
-    int ret;
+    if (ok) *ok = false;
+
+    if (data.isEmpty())
+        return {};
 
-    dest.clear();
+    const int BUFSIZE = 128 * 1024;
+    char tmpBuf[BUFSIZE] = {0};
 
     z_stream strm;
     strm.zalloc = Z_NULL;
     strm.zfree = Z_NULL;
     strm.opaque = Z_NULL;
-    strm.next_in = reinterpret_cast<uchar *>(src.data());
-    strm.avail_in = src.length();
-    strm.next_out = reinterpret_cast<uchar *>(tmpBuf);
+    strm.next_in = reinterpret_cast<const Bytef *>(data.constData());
+    strm.avail_in = uInt(data.size());
+    strm.next_out = reinterpret_cast<Bytef *>(tmpBuf);
     strm.avail_out = BUFSIZE;
 
     // windowBits = 15 + 16 to enable gzip
     // From the zlib manual: windowBits can also be greater than 15 for optional gzip encoding. Add 16 to windowBits
     // to write a simple gzip header and trailer around the compressed data instead of a zlib wrapper.
-    ret = deflateInit2(&strm, Z_BEST_COMPRESSION, Z_DEFLATED, 15 + 16, 8, Z_DEFAULT_STRATEGY);
+    int result = deflateInit2(&strm, level, Z_DEFLATED, (15 + 16), 9, Z_DEFAULT_STRATEGY);
```

Comment:
I mean that you do not need variable assignment at all.
line 62:
`if (deflateInit2(&strm, level, Z_DEFLATED, (15 + 16), 9, Z_DEFAULT_STRATEGY) != Z_OK)`
line 72:
`if (deflate(&strm, Z_NO_FLUSH))`
line 83: 
```
// 'result' is always Z_OK when we reach here, or we exited otherwise
while (deflate(&strm, Z_FINISH) != Z_STREAM_END)
```